app/src/main/java/com/babylon/wallet/android/data/repository/transaction/TransactionRepository.kt

```diff
@@ -41,4 +51,23 @@ class TransactionRepositoryImpl @Inject constructor(private val transactionApi:
     override suspend fun getTransactionPreview(body: TransactionPreviewRequest): Result<TransactionPreviewResponse> {
         return transactionApi.transactionPreview(body).toResult()
     }
+
+    override suspend fun getAccountDepositPreValidation(
+        accountAddress: String,
+        resourceAddress: List<String>
```

Comment:
better `AccountAddress` and `ResourceAddress` ?

Reply:
`String`s  are long since verboten! :) 

The **only** usecase for a String is as a _backing type_ of a newtype which models dynamic data, such as `struct Firstname(String)` - and then for which we will wanna write some validation, e.g. max 255 chars etc.

[Rule #3 of Object Calisthenics - "Wrap all primitives and strings"](https://williamdurand.fr/2013/06/03/object-calisthenics/#3-wrap-all-primitives-and-strings) :)